boards/Makefile.include.cortexm_common

```diff
@@ -1,18 +1,34 @@
 # Target triple for the build. Use arm-none-eabi if you are unsure.
 export TARGET_TRIPLE ?= arm-none-eabi
 
-# Toolchain prefix, defaults to target triple followed by a dash, you will most likely not need to touch this.
+# Use TOOLCHAIN environment variable to select the toolchain to use.
+ifeq (,$(TOOLCHAIN))
+# Default: gnu
+TOOLCHAIN = gnu
+endif
+
+export TOOLCHAIN
+
+ifeq (,$(PREFIX))
+# default toolchain prefix, defaults to target triple followed by a dash, you
+# will most likely not need to touch this.
 export PREFIX ?= $(if $(TARGET_TRIPLE),$(TARGET_TRIPLE)-)
+endif
 
-# we build all cortex boards with the GNU toolchain
-include $(RIOTBOARD)/Makefile.include.gnu
+include $(RIOTBOARD)/Makefile.include.$(TOOLCHAIN)
```

Comment:
moved to above the link test for nano specs, the test requires that LINK is defined